includes/manage_pages.inc

```diff
@@ -568,27 +568,30 @@ function islandora_paged_content_ingest_zipped_pages(AbstractObject $object, $al
  *   Drupal form definition.
  */
 function islandora_paged_content_zipped_upload_form(array $form, array &$form_state, $pid, $allowed_extensions, $model, $derivatives) {
-  module_load_include('inc', 'islandora_ocr', 'includes/utilities');
   module_load_include('inc', 'islandora_paged_content', 'includes/utilities');
   $object = islandora_object_load($pid);
   $current_pages = islandora_paged_content_get_pages($object);
   $last_page_number = count($current_pages);
-  $languages = module_exists('islandora_ocr') ? islandora_ocr_get_enabled_tesseract_languages() : array();
   $upload_size = min((int) ini_get('post_max_size'), (int) ini_get('upload_max_filesize'));
   $extensions = array('zip');
   $form = array();
-  $do_ocr = in_array('ocr', $derivatives) && module_exists('islandora_ocr');
   $message = t("This sequence currently has @count pages. Additional pages will be appended to the end of the sequence by default. !break", array("@count" => $last_page_number, '!break' => '<br />'));
   $message .= t("Choose a number lower than @count to insert page(s) at a specific location in the sequence.", array("@count" => $last_page_number, '!break' => '<br />'));
 
-  $form['language'] = array(
-    '#access' => module_exists('islandora_ocr') && $do_ocr,
-    '#title' => t('Language'),
-    '#type' => 'select',
-    '#description' => t('Please select the language the page is written in.'),
-    '#options' => $languages,
-    '#default_value' => 'English',
-  );
+  if (module_exists('islandora_ocr') && in_array('ocr', $derivatives)) {
+    module_load_include('inc', 'islandora_ocr', 'includes/utilities');
+    $languages = islandora_ocr_get_enabled_tesseract_languages();
+    $form['language'] = array(
+      '#title' => t('Language'),
+      '#type' => 'select',
+      '#description' => t('Please select the language the page is written in.'),
+      '#options' => $languages,
+    );
+    if (in_array('English', $languages)) {
+      $form['language']['#default_value'] = 'eng';
```

Comment:
and if not in array english?  no default, in that case, wonder if we need to make this field required? Does tesseract work without passing a language?

Reply:
I don't think it will work ... but I'd think installing languages would be part of the tesseract installation process anyway. The former method, where it was expecting "English" to be in there, came with the same problem (if it isn't in there, then what's the default?)

An improvement ticket might be in order where the islandora_ocr module itself allows you to define the tesseract default language? Currently it only lets you select which ones are enabled. That would give us a clear path to clean this whole mess right up.